main/Mastermind.rb

```diff
@@ -0,0 +1,135 @@
+class Mastermind
+    COLORS = ["red", "blue", "yellow", "green", "purple", "black"]
+    MAX_ATTEMPTS = 6
+   
+
+
+    def initialize 
+        @num_attempts = 0
+        @user_guess = []
+    end
+
+    def play 
+        print_instructions
+        run_game
+    end
+
+    private
+
+    def print_instructions 
+        puts "Welcome New Mastermind!!"
+        puts "The game is simple, Guess the code and you win. You will have 12 attempts."
+        puts "The code will be made up of four assorted colors."
+        puts "Red, yellow, green, purple, blue and black. As you guess the right color in the right slot,"
+        puts "I'll let you know if your correct. By giving you a hint for what colors are correct. Slowly but surely, this will reveal the color code."
+        puts "If I give you a white dot, then you have a color correct. A black dot, means on of the colors are in correct."
+        puts "First, what will be your first guess?? You will need to type in enter between each color. >>>"
+        
+        puts "-"*20
+        puts "Would you like to attempt to guess the code or have the computer try to beat your selected code?"
+        puts "You guess the code, choose [guesser]. Have the comupter try to beat you, [code master]. "
+    end
+
+    def color_generator
+        4.times.map {COLORS.sample}
+    end
+
+    def input(user_input)
+        guess_slot1 = gets.chomp
+        guess_slot2 = gets.chomp
+        guess_slot3 = gets.chomp
+        guess_slot4 = gets.chomp
+        @user_guess.push(guess_slot1, guess_slot2, guess_slot3, guess_slot4)
+    end
+    
+    
+    def valid_input(user_input)
+        user_input.each do |valid| 
+            if  COLORS.include?(valid)
+                
```

Comment:
It looks like the empty block under the `if` statement here is saying "if the color is valid, do nothing". That works great. Alternatively, this is a place where you could replace the whole  `if/else` with `unless`, since you only need to do something in the `else` branch. e.g.:

````
unless COLORS.include?(valid)
  puts " "*2
  puts user_input.difference(COLORS) 
  puts "^^^ This color/s is not an option. Try not to include them in your next guess."
end
````